Add guidance about optional questions (#2694)
alphagov/govuk-design-system: src/patterns/check-answers/index.md

If a user changes their response in a way that means you need to ask them more questions, do this before returning them to the check answers page.

> If you have questions that are optional, you should let users know that they skipped answering it by showing the question as as ‘Not provided’.

Reviewer: Small suggestion to spell things out a bit more. Looks good either way though.

```suggestion
If you have questions that are optional, let users know that they've skipped it without giving an answer by showing their response as ‘Not provided’.
```
